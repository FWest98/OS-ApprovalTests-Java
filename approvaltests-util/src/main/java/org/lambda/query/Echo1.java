package org.lambda.query;

import com.spun.util.DeprecatedException;
import org.lambda.functions.Function1;

/**
 * @deprecated use lambdas: {@literal  a -> a }
 */
@Deprecated
public class Echo1<T> implements Function1<T, T>
{
  public T call(T i)
  {
    throw new DeprecatedException("a -> a");
  }
}
