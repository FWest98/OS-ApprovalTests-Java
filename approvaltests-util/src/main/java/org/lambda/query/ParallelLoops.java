package org.lambda.query;

import com.spun.util.LambdaThreadLauncher;
import org.lambda.actions.Action1;

import java.util.concurrent.atomic.AtomicInteger;

public class ParallelLoops
{
  public static <T> void forEach(Iterable<T> list, final Action1<T> a1)
  {
    final AtomicInteger done = new AtomicInteger();
    int count = 0;
    for (T i : list)
    {
      count++;
      final T piece = i;
      new LambdaThreadLauncher(() -> {
        a1.call(piece);
        done.incrementAndGet();
      });
    }
    while (done.get() != count)
    {
      try
      {
        Thread.sleep(2);
      }
      catch (InterruptedException e)
      {
        // do nothing
      }
    }
  }
}
