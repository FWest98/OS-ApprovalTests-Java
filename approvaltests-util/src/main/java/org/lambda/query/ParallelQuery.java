package org.lambda.query;

import com.spun.util.LambdaThreadLauncher;
import org.lambda.functions.Function1;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;

public class ParallelQuery
{
  public static <In> List<In> where(Iterable<In> list, final Function1<In, Boolean> funct)
  {
    final ArrayList<In> out = new ArrayList<In>();
    final AtomicInteger done = new AtomicInteger();
    int count = 0;
    for (In i : list)
    {
      count++;
      final In piece = i;
      new LambdaThreadLauncher(() -> {
        if (funct.call(piece))
        {
          out.add(piece);
        }
        done.incrementAndGet();
      });
    }
    while (done.get() != count)
    {
      try
      {
        Thread.sleep(2);
      }
      catch (InterruptedException e)
      {
        // do nothing
      }
    }
    return out;
  }
}
