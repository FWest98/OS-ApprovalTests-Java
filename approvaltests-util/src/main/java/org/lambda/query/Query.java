package org.lambda.query;

import java.util.*;

import org.lambda.functions.Function1;
import org.lambda.query.OrderBy.Order;

import com.spun.util.ArrayUtils;

public class Query<In>
{
  public static <In, Out> Queryable<Out> select(Collection<In> list, Function1<In, Out> function)
  {
    Queryable<Out> out = new Queryable<Out>();
    for (In i : list)
    {
      out.add(function.call(i));
    }
    return out;
  }
  public static <In, Out> Queryable<Out> select(In[] list, Function1<In, Out> function)
  {
    return select(ArrayUtils.asList(list), function);
  }
  public static <In> Queryable<In> where(Iterable<In> list, Function1<In, Boolean> funct)
  {
    Queryable<In> out = new Queryable<In>();
    for (In i : list)
    {
      if (funct.call(i))
      {
        out.add(i);
      }
    }
    return out;
  }
  public static <In> In first(In[] list, Function1<In, Boolean> filter)
  {
    return first(ArrayUtils.asList(list), filter);
  }
  public static <In> In first(Iterable<In> list, Function1<In, Boolean> filter)
  {
    for (In i : list)
    {
      if (filter.call(i))
      { return i; }
    }
    return null;
  }
  public static <In> Queryable<In> where(In[] list, Function1<In, Boolean> filter)
  {
    Queryable<In> out = new Queryable<In>();
    for (In i : list)
    {
      if (filter.call(i))
      {
        out.add(i);
      }
    }
    return out;
  }
  public static <In, Out extends Comparable<Out>> In max(Collection<In> list, Function1<In, Out> f1)
  {
    return getTop(list, f1, 1);
  }
  public static <In, Out extends Comparable<Out>> In min(List<In> list, Function1<In, Out> f1)
  {
    return getTop(list, f1, -1);
  }
  public static <In> Double average(List<In> list, Function1<In, Number> f1)
  {
    double total = 0.00;
    for (In in : list)
    {
      total += f1.call(in).doubleValue();
    }
    return total / list.size();
  }
  private static <In, Out extends Comparable<Out>> In getTop(Collection<In> list, Function1<In, Out> f1,
      int modifier)
  {
    if (ArrayUtils.isEmpty(list))
    { return null; }
    In found = list.iterator().next();
    Out max = f1.call(found);
    for (In in : list)
    {
      Out current = f1.call(in);
      if (max.compareTo(current) * modifier < 0)
      {
        max = current;
        found = in;
      }
    }
    return found;
  }
  public static <T, Out extends Comparable<Out>> T[] orderBy(T[] list, Function1<T, Out> f1)
  {
    return orderBy(list, Order.Ascending, f1);
  }
  public static <T, Out extends Comparable<Out>> T[] orderBy(T[] list, Order order, Function1<T, Out> f1)
  {
    Arrays.sort(list, new OrderBy<T, Out>(order, f1));
    return list;
  }
  public static <T> Queryable<T> orderBy(List<T> list, Function1<T, Comparable<?>> f1)
  {
    return orderBy(list, Order.Ascending, f1);
  }
  public static <T> Queryable<T> orderBy(List<T> list, Order order, Function1<T, Comparable<?>> f1)
  {
    Collections.sort(list, new OrderBy<T, Comparable<?>>(order, f1));
    return Queryable.as(list);
  }
  /**
   * Why does sum() return double? see {@link #sum(Number[])}
   */
  public static <In, Out extends Number> Double sum(In[] list, Function1<In, Out> f1)
  {
    return sum(ArrayUtils.asList(list), f1);
  }
  /**
   * Why does sum() return double? see {@link #sum(Number[])}
   */
  public static <In, Out extends Number> Double sum(Collection<In> list, Function1<In, Out> f1)
  {
    double sum = 0;
    for (In in : list)
    {
      sum += f1.call(in).doubleValue();
    }
    return sum;
  }
  /**
   * Why does sum() return double? see {@link #sum(Number[])}
   */
  public static <Out extends Number> Double sum(Collection<Out> list)
  {
    return sum(list, a -> a);
  }
  /**
   * <H1>Why does sum() return double?</H1><br>
   * sum() needs to handle possible overflow.<br>
   * For example:<br>
   * <code>Queryable.as(100, 100, 100).sum(n -&gt; n.byteValue())</code> should equal 300.<br>
   * Because Double handles the largest range of any number, that is what is returned by default. You can convert
   * it to whichever type you want, but be aware that overflowed values may be truncated.
   */
  public static <Out extends Number> Double sum(Out[] list)
  {
    return sum(list, a -> a);
  }
  @SuppressWarnings({"rawtypes", "unchecked"})
  public static <T extends Number> T max(List<T> numbers)
  {
    return (T) max(numbers, (a) -> (Comparable) a);
  }
  public static <T extends Number> T max(T[] numbers)
  {
    return max(ArrayUtils.asList(numbers));
  }
  @SuppressWarnings({"rawtypes", "unchecked"})
  public static <T extends Number> T min(List<T> numbers)
  {
    return (T) min((List) numbers, (Comparable a) -> a);
  }
  public static <In> boolean all(In[] array, Function1<In, Boolean> funct)
  {
    return array.length == where(array, funct).size();
  }
  public static <In> boolean all(List<In> array, Function1<In, Boolean> funct)
  {
    return array.size() == where(array, funct).size();
  }
  public static <In> boolean any(List<In> array, Function1<In, Boolean> funct)
  {
    return first(array, funct) != null;
  }
  public static <In> boolean any(In[] array, Function1<In, Boolean> funct)
  {
    return first(array, funct) != null;
  }
  public static <In> Queryable<In> distinct(List<In> list)
  {
    Queryable<In> distinct = new Queryable<>();
    for (In in : list)
    {
      if (!distinct.contains(in))
      {
        distinct.add(in);
      }
    }
    return distinct;
  }
  public static <In> In last(In[] list)
  {
    return last(Arrays.asList(list));
  }
  static <In> In last(List<In> asList)
  {
    if (asList.isEmpty())
    { return null; }
    return asList.get(asList.size() - 1);
  }
  public static <Out, In> Queryable<Out> selectMany(Queryable<In> list, Function1<In, Collection<Out>> selector)
  {
    Queryable<Out> out = new Queryable<Out>();
    for (In i : list)
    {
      out.addAll(selector.call(i));
    }
    return out;
  }
  public static <Out, In> Queryable<Out> selectManyArray(Queryable<In> list, Function1<In, Out[]> selector)
  {
    Queryable<Out> out = new Queryable<Out>();
    for (In i : list)
    {
      out.addAll(Arrays.asList(selector.call(i)));
    }
    return out;
  }
}
