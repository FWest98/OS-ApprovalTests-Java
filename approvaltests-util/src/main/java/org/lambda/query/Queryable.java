package org.lambda.query;

import com.spun.util.ClassUtils;
import com.spun.util.ObjectUtils;
import org.lambda.Extendable;
import org.lambda.functions.Function1;
import org.lambda.query.OrderBy.Order;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import java.util.Map.Entry;

public class Queryable<In> extends ArrayList<In>
{
  // TODO: autogenerate this;
  private final long serialVersionUID = 1L;
  private Class<In>  type;
  public Queryable()
  {
    this(null);
  }
  public Queryable(Class<In> type)
  {
    this.type = type;
  }
  public <T extends Extendable<List<In>>> T use(Class<T> that)
  {
    try
    {
      T t = that.newInstance();
      t.setCaller(this);
      return t;
    }
    catch (Throwable t)
    {
      throw ObjectUtils.throwAsError(t);
    }
  }
  public <Out> Queryable<Out> select(Function1<In, Out> function)
  {
    return Query.select(this, function);
  }
  public Queryable<In> where(Function1<In, Boolean> funct)
  {
    return Query.where(this, funct);
  }
  public In first()
  {
    return firstOrDefault(null);
  }
  public In first(Function1<In, Boolean> filter)
  {
    return Query.first(this, filter);
  }
  public In firstOrDefault(In defaultValue)
  {
    return this.isEmpty() ? defaultValue : this.get(0);
  }
  public boolean all(Function1<In, Boolean> filter)
  {
    return Query.all(this, filter);
  }
  public <Out extends Comparable<Out>> In max(Function1<In, Out> f1)
  {
    return Query.max(this, f1);
  }
  public <Out extends Comparable<Out>> In min(Function1<In, Out> f1)
  {
    return Query.min(this, f1);
  }
  public Double average(Function1<In, Number> f1)
  {
    return Query.average(this, f1);
  }
  public Queryable<In> orderBy(Function1<In, Comparable<?>> f1)
  {
    return Query.orderBy(this, f1);
  }
  public Queryable<In> orderBy(Order order, Function1<In, Comparable<?>> f1)
  {
    return Query.orderBy(this, order, f1);
  }
  /**
   * Why does sum() return double? see {@link org.lambda.query.Query#sum(Number[])}
   */
  public <Out extends Number> Double sum(Function1<In, Out> f1)
  {
    return Query.sum(this, f1);
  }
  /**
   * Why does sum() return double? see {@link org.lambda.query.Query#sum(Number[])}
   */
  public Double sum()
  {
    return Query.sum((List<Number>) this);
  }
  public In max()
  {
    return (In) Query.max((List<Number>) this);
  }
  public In min()
  {
    return (In) Query.min((List<Number>) this);
  }
  public boolean any(Function1<In, Boolean> funct)
  {
    return Query.any(this, funct);
  }
  public static <T> Queryable<T> as(List<T> list)
  {
    Class<?> type = ClassUtils.getGreatestCommonBaseType(list);
    return Queryable.as(list, (Class<T>) type);
  }
  public static <T> Queryable<T> as(List<T> list, Class<T> type)
  {
    if (list instanceof Queryable)
    { return (Queryable<T>) list; }
    Queryable<T> q = new Queryable<T>(type);
    q.addAll(list);
    return q;
  }
  public static <T> Queryable<T> as(T... array)
  {
    return as(Arrays.asList(array), (Class<T>) array.getClass().getComponentType());
  }
  /**
   * Maintains order
   */
  public Queryable<In> distinct()
  {
    return Query.distinct(this);
  }
  public In[] asArray()
  {
    int size = this.size();
    In[] result = (In[]) Array.newInstance(getType(), size);
    return toArray(result);
  }
  public Class<In> getType()
  {
    if (type == null)
    {
      type = (Class<In>) ClassUtils.getGreatestCommonBaseType(this);
    }
    return type;
  }
  public <Out> Queryable<Out> selectMany(Function1<In, Collection<Out>> selector)
  {
    return Query.selectMany(this, selector);
  }
  public <Out> Queryable<Out> selectManyArray(Function1<In, Out[]> selector)
  {
    return Query.selectManyArray(this, selector);
  }
  public <Key> Queryable<Entry<Key, Queryable<In>>> groupBy(Function1<In, Key> keySelector)
  {
    return Query.groupBy(this, keySelector);
  }
  public <Key, Out1, Out2> Queryable<Entry<Key, Out2>> groupBy(Function1<In, Key> keySelector,
      Function1<In, Out1> valueSelector, Function1<Queryable<Out1>, Out2> resultSelector)
  {
    return Query.groupBy(this, keySelector, valueSelector, resultSelector);
  }
  public String join(String joinCharacter)
  {
    return String.join(joinCharacter, this.select(t -> "" + t));
  }
  public <Out> String join(String joinCharacter, Function1<In, Out> transformer)
  {
    return String.join(joinCharacter, this.select(t -> "" + transformer.call(t)));
  }
}
