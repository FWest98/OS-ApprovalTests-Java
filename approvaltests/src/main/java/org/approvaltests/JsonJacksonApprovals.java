package org.approvaltests;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.spun.util.ObjectUtils;
import org.approvaltests.core.Options;

public class JsonJacksonApprovals
{
  public static void verifyAsJson(Object o)
  {
    verifyAsJson(o, new Options());
  }
  public static void verifyAsJson(Object o, Options options)
  {
    Approvals.verify(asJson(o), options.forFile().withExtension(".json"));
  }
  public static String asJson(Object o)
  {
    try
    {
      ObjectMapper objectMapper = new ObjectMapper();
      return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(o);
    }
    catch (JsonProcessingException e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
}
