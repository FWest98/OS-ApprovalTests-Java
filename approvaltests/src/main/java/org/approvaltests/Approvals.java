package org.approvaltests;

import com.spun.util.*;
import com.spun.util.persistence.ExecutableQuery;
import com.spun.util.persistence.Loader;
import com.spun.util.persistence.SqlLoader;
import org.approvaltests.approvers.ApprovalApprover;
import org.approvaltests.approvers.FileApprover;
import org.approvaltests.core.ApprovalFailureReporter;
import org.approvaltests.core.ApprovalWriter;
import org.approvaltests.namer.ApprovalNamer;
import org.approvaltests.namer.StackTraceNamer;
import org.approvaltests.reporters.ExecutableQueryFailure;
import org.approvaltests.writers.*;
import org.lambda.actions.Action0;
import org.lambda.functions.Function1;
import org.lambda.query.Query;

import java.io.File;
import java.io.FileFilter;
import java.io.FilenameFilter;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

public class Approvals
{
  public static Loader<ApprovalNamer> namerCreater = new Loader<ApprovalNamer>()
  {
    public ApprovalNamer load()
    {
      return new StackTraceNamer();
    }
  };
  public static void verify(String response)
  {
    verify(new ApprovalTextWriter(response, "txt"));
  }
  public static void verify(Object o)
  {
    verify("" + o);
  }
  public static <T> void verifyAll(String label, T[] array)
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"));
  }
  public static <T> void verifyAll(String header, String label, T[] array)
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"));
  }
  public static <T> void verifyAll(T[] values, Function1<T, String> f1)
  {
    String text = ArrayUtils.toString(values, f1);
    verify(new ApprovalTextWriter(text, "txt"));
  }
  public static <T> void verifyAll(String header, T[] values, Function1<T, String> f1)
  {
    verifyAll(header, Arrays.asList(values), f1);
  }
  public static <T> void verifyAll(String header, Iterable<T> array, Function1<T, String> f1)
  {
    String text = formatHeader(header) + ArrayUtils.toString(array, f1);
    verify(new ApprovalTextWriter(text, "txt"));
  }
  private static String formatHeader(String header)
  {
    return StringUtils.isEmpty(header) ? "" : header + "\n\n\n";
  }
  public static <T> void verifyAll(String label, Iterable<T> array)
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"));
  }
  public static <T> void verifyAll(String header, String label, Iterable<T> array)
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"));
  }

  public static void verifyHtml(String response)
  {
    verify(new ApprovalTextWriter(response, "html"));
  }
  public static void verify(File generateFile)
  {
    verify(new FileApprovalWriter(generateFile));
  }

  public static void verify(ApprovalWriter writer, ApprovalNamer namer, ApprovalFailureReporter reporter)
  {
    verify(new FileApprover(writer, namer), reporter);
  }
  public static void verify(ApprovalWriter writer)
  {
    verify(writer, createApprovalNamer(), getReporter());
  }
  public static void verifyXml(String xml)
  {
    verify(new ApprovalXmlWriter(xml));
  }
  public static void verify(ApprovalApprover approver)
  {
    verify(approver, getReporter());
  }
  public static void verify(ApprovalApprover approver, ApprovalFailureReporter reporter)
  {
    if (!approver.approve())
    {
      approver.reportFailure(reporter);
      approver.fail();
    }
    else
    {
      approver.cleanUpAfterSuccess(reporter);
    }
  }
  public static void verify(ExecutableQuery query)
  {
    verify(new ApprovalTextWriter(query.getQuery(), "txt"), createApprovalNamer(),
        new ExecutableQueryFailure(query));
  }
  public static void verify(Map<?, ?> map)
  {
    verify(new ApprovalTextWriter(StringUtils.toString(map), "txt"));
  }
  public static void verify(ResultSet rs)
  {
    verify(new ResultSetApprovalWriter(rs));
  }
  public static <T> void verify(SqlLoader<T> loader)
  {
    verify(new SqlLoader.ExecutableWrapper<T>(loader));
  }
  public static ApprovalNamer createApprovalNamer()
  {
    return namerCreater.load();
  }

  public static void verifyEachFileInDirectory(File directory)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles());
  }
  public static void verifyEachFileInDirectory(File directory, FileFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  public static void verifyEachFileInDirectory(File directory, FilenameFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  private static void verifyEachFileAgainstMasterDirectory(File[] files) throws Error
  {
    ApprovalNamer namer = createApprovalNamer();
    String dirName = namer.getSourceFilePath() + File.separator + namer.getApprovalName() + ".Files";
    File approvedDirectory = new File(dirName);
    List<File> mismatched = new ArrayList<File>();
    for (File f : files)
    {
      if (!f.isDirectory())
      {
        try
        {
          verify(new DirectoryToDirectoryWriter(f, approvedDirectory));
        }
        catch (Throwable e)
        {
          mismatched.add(f);
        }
      }
    }
    if (!mismatched.isEmpty())
    {
      String message = "The Following Files did not match up: " + getFileNameList(mismatched);
      throw new Error(message);
    }
  }
  private static String getFileNameList(List<File> mismatched)
  {
    return Query.select(mismatched, a -> a.getName()).toString();
  }
  public static void verifyJson(String json)
  {
    verify(JsonUtils.prettyPrint(json), "json");
  }
  public static void verify(Object actual, String fileExtensionWithoutDot)
  {
    verify(new ApprovalTextWriter("" + actual, fileExtensionWithoutDot));
  }
  public static ApprovalFailureReporter getReporter()
  {
    return ReporterFactory.get();
  }
  public static void verifyAsJson(Object o)
  {
    verify(JsonUtils.asJson(o), "json");
  }
  public static void verifyException(Action0 runnableBlock)
  {
    Throwable t = ObjectUtils.captureException(runnableBlock);
    if (t == null) { throw new FormattedException("No exception thrown when running %s", runnableBlock); }
    Approvals.verify(String.format("%s: %s", t.getClass().getName(), t.getMessage()));
  }
}