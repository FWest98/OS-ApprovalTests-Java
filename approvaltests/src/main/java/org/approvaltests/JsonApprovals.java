package org.approvaltests;

import com.google.gson.GsonBuilder;
import com.spun.util.*;
import com.spun.util.persistence.ExecutableCommand;
import com.spun.util.persistence.Loader;
import com.spun.util.persistence.SqlLoader;
import org.approvaltests.approvers.ApprovalApprover;
import org.approvaltests.approvers.FileApprover;
import org.approvaltests.core.ApprovalFailureReporter;
import org.approvaltests.core.ApprovalWriter;
import org.approvaltests.core.Options;
import org.approvaltests.core.VerifyResult;
import org.approvaltests.namer.ApprovalNamer;
import org.approvaltests.namer.MasterDirectoryNamer;
import org.approvaltests.namer.StackTraceNamer;
import org.approvaltests.reporters.ExecutableQueryFailure;
import org.approvaltests.writers.ApprovalXmlWriter;
import org.lambda.actions.Action0;
import org.lambda.functions.Function1;
import org.lambda.query.Query;

import java.io.File;
import java.io.FileFilter;
import java.io.FilenameFilter;
import java.sql.ResultSet;
import java.util.*;

/**
 * Requires (GSON)[https://mvnrepository.com/artifact/com.google.code.gson/gson]
 */
public class JsonApprovals
{
  public static void verifyJson(String json)
  {
    verifyJson(json, new Options());
  }
  public static void verifyJson(String json, Options options)
  {
    Approvals.verify(JsonUtils.prettyPrint(json), options.forFile().withExtension(".json"));
  }
  public static void verifyAsJson(Object o)
  {
    verifyAsJson(o, new Options());
  }
  public static void verifyAsJson(Object o, Function1<GsonBuilder, GsonBuilder> gsonBuilder)
  {
    verifyAsJson(o, gsonBuilder, new Options());
  }
  public static void verifyAsJson(Object o, Function1<GsonBuilder, GsonBuilder> gsonBuilder, Options options)
  {
    Approvals.verify(JsonUtils.asJson(o, gsonBuilder),
        options.forFile().withExtension(".json"));
  }
  public static void verifyAsJson(Object o, Options options)
  {
    Approvals.verify(JsonUtils.asJson(o), options.forFile().withExtension(".json"));
  }
}
