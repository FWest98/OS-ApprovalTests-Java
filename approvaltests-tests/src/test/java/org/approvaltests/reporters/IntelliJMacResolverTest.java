package org.approvaltests.reporters;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.lambda.query.Queryable;

class IntelliJMacResolverTest
{
  @Test
  void testFindIt()
  {
    Queryable<String> q = new Queryable<>(String.class);
    IntelliJMacResolver.getDiffInfo("user/lars", f -> {
      q.add(f);
      return false;
    });
    Approvals.verifyAll("Files", q);
  }
}
