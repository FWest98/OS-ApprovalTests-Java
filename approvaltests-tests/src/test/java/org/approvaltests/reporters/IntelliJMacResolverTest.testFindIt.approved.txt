Files[0] = /Applications/IntelliJ IDEA Ultimate.app/Contents/MacOS/idea
Files[1] = /Applications/IntelliJ IDEA.app/Contents/MacOS/idea
Files[2] = /Applications/IntelliJ IDEA Community.app/Contents/MacOS/idea
Files[3] = /Applications/IntelliJ IDEA Community Edition.app/Contents/MacOS/idea
Files[4] = user/lars/Applications/IntelliJ IDEA Ultimate.app/Contents/MacOS/idea
Files[5] = user/lars/Applications/IntelliJ IDEA.app/Contents/MacOS/idea
Files[6] = user/lars/Applications/IntelliJ IDEA Community.app/Contents/MacOS/idea
Files[7] = user/lars/Applications/IntelliJ IDEA Community Edition.app/Contents/MacOS/idea
